File: src/main/java/net/shadowfacts/underwaterutilities/util/energy/FUAdapter.java
package net.shadowfacts.underwaterutilities.util.energy;

import net.darkhax.tesla.api.implementation.BaseTeslaContainer;
import net.minecraftforge.energy.IEnergyStorage;

/**
 * @author shadowfacts
 */
public class FUAdapter implements IEnergyStorage {

	private final BaseTeslaContainer tesla;

	public FUAdapter(BaseTeslaContainer tesla) {
		this.tesla = tesla;
	}

	@Override
	public int receiveEnergy(int maxReceive, boolean simulate) {
		return (int)tesla.givePower(maxReceive, simulate);
	}

	@Override
	public int extractEnergy(int maxExtract, boolean simulate) {
		return (int)tesla.takePower(maxExtract, simulate);
	}

	@Override
	public int getEnergyStored() {
		return (int)tesla.getStoredPower();
	}

	@Override
	public int getMaxEnergyStored() {
		return (int)tesla.getCapacity();
	}

	@Override
	public boolean canExtract() {
		return tesla.getOutputRate() > 0;
	}

	@Override
	public boolean canReceive() {
		return tesla.getInputRate() > 0;
	}
	
}

File: src/main/kotlin/net/shadowfacts/underwaterutilities/util/energy/FUAdapter.kt
package net.shadowfacts.underwaterutilities.util.energy

import net.darkhax.tesla.api.implementation.BaseTeslaContainer
import net.minecraftforge.energy.IEnergyStorage

/**
 * @author shadowfacts
 */
class FUAdapter(private val tesla: BaseTeslaContainer): IEnergyStorage {

	override fun receiveEnergy(maxReceive: Int, simulate: Boolean): Int {
		return tesla.givePower(maxReceive.toLong(), simulate).toInt()
	}

	override fun extractEnergy(maxExtract: Int, simulate: Boolean): Int {
		return tesla.takePower(maxExtract.toLong(), simulate).toInt()
	}

	override fun getEnergyStored() = tesla.storedPower.toInt()

	override fun getMaxEnergyStored() = tesla.capacity.toInt()

	override fun canExtract() = tesla.outputRate > 0

	override fun canReceive() = tesla.inputRate > 0

}
